controllers/module_reconciler.go

```diff
@@ -256,9 +271,19 @@ func (r *ModuleReconciler) handleBuild(ctx context.Context,
 	km *kmmv1beta1.KernelMapping,
 	kernelVersion string,
 	containerImage string) (bool, error) {
-	if mod.Spec.ModuleLoader.Container.Build == nil && km.Build == nil {
+
+	//if theres no build specified skip this section
+	if !r.willBuild(mod, km) {
 		return false, nil
 	}
+	//if build is specified AND sign is specified then we want to build an intermediate image
+	// and let sign produce the one containerImage specifies
+	if r.willSign(mod, km) {
+		containerImage = r.appendToTag(containerImage, "unsigned")
+	}
```

Comment:
@qbarrand , i think adding an output field to the build section might be confusing. In case there is no sign section, the output should be the containerImage, but in case a user have set the output field - it will cause confusion. What about making the UnsignedImage field mandatory in the Sign section?  It will also mean the name of the output image for Build section in case Sign section is present

Reply:
That sounds reasonable, but still a bit magic. We wanted to have build and sign completely independent, and now the build behaviour would depend on whether a `sign` section is present or not.
I think I would rather have a `build.output` field identical to `sign.unsignedImage`, and have the webhook enforce that. That means some repetition, but at least it makes `build` and `sign` completely independent from each other. @chr15p WDYT?

Reply:
Firstly (and trivially) I dont like the name `build.Output`, I think it should be `build.OutputImage` because its intent is a bit clearer and matches the naming style of `sign.UnsignedImage`.

If we say that build produces `sign.UnsignedImage` as the intermediate image name, then what happens if you define multiple `km.build` sections but a single default `Spec.ModuleLoader.Container.Sign` because you want kmods for various kernels but all signed with the same key pair? that feels like a very common pattern that would be very hard to handle if we 
reuse `sign.UnsignedImage`

Using `build.OutputImage` would get around this because each kernelmapping would produce a different image.

Whether `sign.UnsignedImage` and/or `build.OutputImage` are defined the final output image should always be the one named in `containerImage`

`containerImage` should always be what we produce and what we run, so (I think)

- If only `build` is defined it should produce `containerImage`

- If only `sign` is present it should consume `sign.UnsignedImage`, and produce `containerImage`

- If `build` and `sign` are present then build produces `build.OutputImage`, and `sign` consumes `build.OutputImage` and produces `containerImage`

Things that are defined but not needed get ignored (so if you define `build.OutputImage` and `sign.UnsignedImage` only OutputImage is used).  This might be a bit confusing for the user at first glance but the docs can explain it fairly easily.

The actions of build and sign are always going to have to depend on whether the other is present to know what the intermediate image is, unless build produces `containerImage` and sign overwrites it, but that has its own issues (single build, multiple signing falls apart for a start).

Thoughts?

Reply:
Its this kind of complexity I was trying to avoid by automatically generating the intermediate image name, but I agree that the small but real chance of clobbering an existing image in the registry is A Bad Thing.

Reply:
following much discussion I have changed the intermediate imagename to 
`imagename:existingtag + "_" + namespace + "_" + modulename + "_kmm_unsigned"`